Cargo.toml

```diff
@@ -95,7 +95,7 @@ maybe-rayon = { version = "0.1.0", default-features = false }
 rayon = "1.5"
 
 # Protobuf and gRPC
-prost = "=0.12.3"
+prost = "0.12.6"
```

Comment:
maybe this is the source of CI fail?

Reply:
It turns out that's not the case. I created a new branch from `zsa1` and made only this single `prost` version change there. I committed it and created a new PR - the `cargo vet` CI check passed without any issues. This can even be verified without a PR: after I made that change in the repository, I ran `cargo check` and `git diff Cargo.lock`- the `Cargo.lock` file wasn't changed.

However, when I uncommented `zcash_client_backend` in `Cargo.toml` and then ran `git diff Cargo.lock`, it started to show a lot of differences like:

```diff
+[[package]]
+name = "anyhow"
+version = "1.0.89"
+source = "registry+https://github.com/rust-lang/crates.io-index"
+checksum = "86fdf8605db99b54d3cd748a44c6d04df638eb5dafb219b135d0149bd0db01f6"
...
```

This means that adding `zcash_client_backend` causes the `cargo vet` issues in CI that we observed.

I guess this happens because `zcash_client_backend` uses additional dependencies in its `Cargo.toml` that require newer versions of sub-dependencies like `anyhow`, which are not marked as audited for `cargo vet1 in our git branch (the audited list is contained in the `supply-chain` folder of the repository).

I think the reason for this is that the maintainers of the original `librustzcash` repo maintain consistency between the `Cargo.toml` files of the crates, the `Cargo.lock` in the root, and the audited list in the `supply-chain` folder. But in our fork, we have copies of `zcash_client_backend/Cargo.toml` and the `supply-chain folder` from different versions of the original branches. I'm not sure though - we need to check the history of our commits to find the source of the difference.

Additionally, I tried copying the `supply-chain` folder from the upstream version (https://github.com/zcash/librustzcash/tree/main/supply-chain), but it didn't resolve the issue. Maybe I missed something - I'm continuing to work on it.

Reply:
I managed to solve the `cargo vet` issue. Here's what I did:

1. Ran `git log zcash_client_backend/Cargo.toml` and found the latest commit message:

> Zsa1 is updated to a more current upstream version, commit:
> c3eace4982dfea82f89758f40c1db2e82a138e98

2. Switched to commit [`c3eace4982dfea82f89758f40c1db2e82a138e98`](https://github.com/zcash/librustzcash/commit/c3eace4982dfea82f89758f40c1db2e82a138e98) (which corresponds to [zcash/librustzcash#1383](https://github.com/zcash/librustzcash/pull/1383) in the original repo).

3. Took the `Cargo.lock` file from that commit and placed it into the `zebra-compat-enable-backend` branch.

4. Ran `cargo check`, `cargo test`, `cargo build` locally, and also `cargo vet --locked` locally to ensure everything works.

5. Committed the changes — now all CI checks have passed.

Summary: I used the existing set of audited dependencies, but had to use an older version of `Cargo.lock` (updated by `cargo check` to include our QED-it-specific dependency crates like `orchard`, etc.)

It seems the issue was due to differences in the `Cargo.lock` and the audited dependencies between our fork and the upstream repository. By aligning our `Cargo.lock` with the one from the relevant upstream commit, we can ensure compatibility and pass the CI checks.